File: backend/app/schemas/review.py
from pydantic import BaseModel
from typing import List, Literal


class ReviewRequest(BaseModel) :
    code : str
    language : str

class Issue(BaseModel) :
    line_number : int
    severity : Literal["high", "medium", "low"]
    category : Literal["bug", "security", "performance", "style"]
    description : str
    suggestion : str
    fixed_code_snippet : str

class ReviewResponse(BaseModel) :
    issues : List[Issue]
    score : int
    summary : str

class ApplyRequest(BaseModel) :
    original_code : str
    language : str
    issue_description : str
    suggestion : str
    fixed_code_snippet : str

class ApplyResponse(BaseModel):
    fixed_code: str
    message: str
    

File: frontend/src/types/review.ts
export type Severity = 'high' | 'medium' | 'low'
export type IssueCategory = 'bug' | 'security' | 'performance' | 'style'

export type ReviewIssue = {
  line_number: number
  severity: Severity
  category: IssueCategory
  description: string
  suggestion: string
  fixed_code_snippet: string
}

export type ReviewRequest = {
  code: string
  language: string
}

export type ReviewResponse = {
  issues: ReviewIssue[]
  score: number
  summary: string
}

export type ApplyFixRequest = {
  original_code: string
  language: string
  issue_description: string
  suggestion: string
  fixed_code_snippet: string
}

export type ApplyFixResponse = {
  fixed_code: string
  message: string
}
